jupyterhub/templates/hub/_helpers-passwords.tpl

```diff
@@ -0,0 +1,67 @@
+{{- /*
+    This file contains logic to lookup already
+    generated passwords or generate a new.
+
+    proxy.secretToken       / hub.config.JupyterHub.proxy_auth_token
+    hub.cookieSecret        / hub.config.JupyterHub.cookie_secret
+    auth.state.cryptoKey    / hub.config.CryptKeeper.keys
+*/}}
+
+{{/*
+    Returns given number of random Hex characters.
+
+    In practice, it generates up to 100 randAlphaNum strings
+    that are filtered from non-hex characters and augmented
+    to the resulting string that is finally trimmed down.
+*/}}
+{{- define "jupyterhub.randHex" -}}
+    {{- $result := "" }}
+    {{- range $i := until 100 }}
+        {{- if lt (len $result) . }}
+            {{- $rand_list := randAlphaNum . | splitList "" -}}
```

Comment:
For those curious - randAlphaNum *is* cryptographically safe too - http://masterminds.github.io/sprig/strings.html.